### PR TITLE
implement gradle plugin to facilitate okhttp instrumentation

### PR DESCRIPTION
**Pull request by snowp**
This adds a gradle plugin inspired by https://github.com/getsentry/sentry-android-gradle-plugin that allows us to automatically install the OkHttpEventListener for all Okhttp clients.

### CODE REVIEW

File: platform/jvm/capture-plugin/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+ plugins {
+    alias(libs.plugins.kotlin)
+
+    id("dependency-license-config")
+    id("java-gradle-plugin")
+    id("maven-publish")
+    id("com.vanniktech.maven.publish") version "0.28.0" apply false
```
Comment (murki) [inline platform/jvm/capture-plugin/build.gradle.kts:7]: you can just reference the libs catalog we use everywhere `alias(libs.plugins.maven.publish)`

File: platform/jvm/capture-plugin/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    plugins {
+        create("capturePlugin") {
+            id = "io.bitdrift.capture.capture-plugin"
+            implementationClass = "io.bitdrift.capture.CapturePlugin"
+        }
+    }
+}
+
+apply {
+    plugin("com.vanniktech.maven.publish")
```
Comment (murki) [inline platform/jvm/capture-plugin/build.gradle.kts:34]: not sure why it doesn't apply it at top just to do it here, maybe some weird ordering issue?

File: platform/jvm/capture-plugin/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+ plugins {
+    alias(libs.plugins.kotlin)
+
+    id("dependency-license-config")
+    id("java-gradle-plugin")
+    id("maven-publish")
```
Comment (murki) [inline platform/jvm/capture-plugin/build.gradle.kts:6]: is this required?

File: platform/jvm/capture-plugin/src/main/kotlin/io/bitdrift/capture/instrumentation/util/ConstantPoolHelpers.kt [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+ * See https://github.com/getsentry/sentry-android-gradle-plugin/issues/360
+ * and https://github.com/getsentry/sentry-android-gradle-plugin/issues/359#issuecomment-1193782500
+ */
+/* ktlint-disable max-line-length */
+private val MINIFIED_CLASSNAME_REGEX = """^(((([a-zA-z])\4{1,}|[a-zA-Z]{1,2})([0-9]{1,})?(([a-zA-Z])\7{1,})?)|([a-zA-Z]([0-9])?))(${'\\'}${'$'}((((\w)\14{1,}|[a-zA-Z]{1,2})([0-9]{1,})?(([a-zA-Z])\17{1,})?)|(\w([0-9])?)))*${'$'}""".toRegex()
+
+/**
+ * See https://github.com/getsentry/sentry/blob/c943de2afc785083554e7fdfb10c67d0c0de0f98/static/app/components/events/eventEntries.tsx#L57-L58
+ */
+private val MINIFIED_CLASSNAME_SENTRY_REGEX =
```
Comment (murki) [inline platform/jvm/capture-plugin/src/main/kotlin/io/bitdrift/capture/instrumentation/util/ConstantPoolHelpers.kt:122]: const name

File: platform/jvm/capture/build.gradle.kts [hunk 5]
```diff
@@ -85,6 +85,8 @@ cargo {
     targetDirectory = "../../../target"
     targets = listOf("arm64", "x86_64")
     pythonCommand = "python3"
+    rustcCommand = "/Users/snow/.cargo/bin/rustc"
+    cargoCommand = "/Users/snow/.cargo/bin/cargo"
```
Comment (murki) [inline platform/jvm/capture/build.gradle.kts:89]: ?

File: platform/jvm/gradle-test-app/src/main/java/io/bitdrift/gradletestapp/FirstFragment.kt [hunk 6]
```diff
@@ -147,7 +147,6 @@ class FirstFragment : Fragment() {
 
     private fun provideOkHttpClient(): OkHttpClient {
         return OkHttpClient.Builder()
-            .eventListenerFactory(CaptureOkHttpEventListenerFactory())
```
Comment (murki) [inline platform/jvm/gradle-test-app/src/main/java/io/bitdrift/gradletestapp/FirstFragment.kt:150]: If there are multiple okhttp clients will it affect them all? e.g. if they use a separate one for apollo graphql

File: platform/jvm/gradle-test-app/src/main/java/io/bitdrift/gradletestapp/MainActivity.kt [hunk 7]
```diff
@@ -16,6 +16,7 @@ import androidx.navigation.ui.AppBarConfiguration
 import androidx.navigation.ui.navigateUp
 import androidx.navigation.ui.setupActionBarWithNavController
 import io.bitdrift.gradletestapp.databinding.ActivityMainBinding
+import okhttp3.OkHttpClient
```
Comment (murki) [inline platform/jvm/gradle-test-app/src/main/java/io/bitdrift/gradletestapp/MainActivity.kt:19]: remove?

File: platform/jvm/gradle/wrapper/gradle-wrapper.properties [hunk 8]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.9-bin.zip
```
Comment (murki) [inline platform/jvm/gradle/wrapper/gradle-wrapper.properties:3]: let's make sure to also update bazel

**Review by murki — APPROVED** [review]

File: platform/jvm/capture-plugin/build.gradle.kts (hunk 2 above)
Comment (snowp) [reply platform/jvm/capture-plugin/build.gradle.kts:34]: Cargo culting the upstream repo, let me see if i can simplify

File: platform/jvm/capture-plugin/build.gradle.kts (hunk 3 above)
Comment (snowp) [reply platform/jvm/capture-plugin/build.gradle.kts:6]: nah, removed

File: platform/jvm/gradle-test-app/src/main/java/io/bitdrift/gradletestapp/FirstFragment.kt (hunk 6 above)
Comment (snowp) [reply platform/jvm/gradle-test-app/src/main/java/io/bitdrift/gradletestapp/FirstFragment.kt:150]: Yeah the plugin will traverse all the clients in the system and add it everywhere. If using both at the same time causes issues we should figure out how to make the interoperate well, this is probably easier than trying to figure out how to make it conditional

File: platform/jvm/gradle/wrapper/gradle-wrapper.properties (hunk 8 above)
Comment (snowp) [reply platform/jvm/gradle/wrapper/gradle-wrapper.properties:3]: can do this separately

File: platform/jvm/gradle-test-app/src/main/java/io/bitdrift/gradletestapp/FirstFragment.kt (hunk 6 above)
Comment (murki) [reply platform/jvm/gradle-test-app/src/main/java/io/bitdrift/gradletestapp/FirstFragment.kt]: ok i'm working on that pr now, i'll think about it

**Pull request merged**